src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -423,12 +423,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ItemGroup>
       <OffByDefaultAnalyzer Include="Microsoft.AspNetCore.Http.RequestDelegateGenerator"
                             IsEnabled="$(EnableRequestDelegateGenerator)"/>
+      <OffByDefaultAnalyzer Include="Microsoft.Extensions.Configuration.Binder.SourceGeneration"
+                            IsEnabled="$(EnableConfigurationBindingGenerator)"/>
     </ItemGroup>
 
     <ItemGroup>
       <AnalyzerToRemove Include="@(OffByDefaultAnalyzer)"
                         Path="%(Analyzer.Identity)"
-                        Condition="'%(Analyzer.Filename)' == '@(OffByDefaultAnalyzer)'" />
+                        Condition="'%(Analyzer.Filename)' == '%(OffByDefaultAnalyzer.Identity)'" />
```

Comment:
I usually do the following: https://github.com/dotnet/runtime/blob/d414646a1e8e400b9e0b06d5ddebae0069f28199/Directory.Build.targets#L192-L197

1. Create a new item with the filename as the identity from the off by default analyzers. Save the original identity as an item attribute. Subtract those from `Analyzer` and the subtract that again from `Analyzer` to just get the analyzers that are off by default from the original `Analyzer` item.

Reply:
@ViktorHofer's example is pretty good.  I had tried something similar that worked for me: https://gist.github.com/ericstj/9c32cb2a365791d86a891cf8d7fb4150.  Probably there is a way to simplify.

Reply:
If you can change the `Include` of the OffByDefaultAnalyzer items to end in `.dll`, you can use `MatchOnMetadata`: https://gist.github.com/ericstj/9c32cb2a365791d86a891cf8d7fb4150?permalink_comment_id=4539791#gistcomment-4539791.

If that's not desirable, then the intermediate-list approaches are the way I'd go too.

Reply:
That's a pretty slick feature, I wasn't aware of that.  https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-items?view=vs-2022#matchonmetadata-attribute  I think that's an elegant solution here, and I don't think it's a problem to add the `.dll` extension to the `OffByDefaultAnalyzer` items since they're expected to be internal to the targets.

Reply:
Thanks folks for helping land on a solution here.